core/src/loader.rs

```diff
@@ -1604,38 +1609,116 @@ impl<'gc> Loader<'gc> {
+            // (we previously had a fake empty SwfMovie).
+            // However, we still use `LoaderStream::NotYetLoaded`, since
+            // the actual MovieClip display object has not run its first
+            // frame yet.
+            loader_info.set_loader_stream(
+                LoaderStream::NotYetLoaded(movie.clone().unwrap(), Some(dobj.unwrap()), false),
+                uc.gc_context,
+            );
+        }
+
+        if let Some(dobj) = dobj {
+            if dobj.as_movie_clip().is_some() {
+                // We call these methods after we initialize the `LoaderInfo`, but before the
+                // add the the loaded clip as a child. The frame constructor should see
+                // 'this.parent == null' and 'this.stage == null'
+                dobj.post_instantiation(uc, None, Instantiator::Movie, false);
+                catchup_display_object_to_frame(uc, dobj);
+                // Movie clips created from ActionScript (including from a Loader) skip the next enterFrame,
+                // and consequently are observed to have their currentFrame lag one
+                // frame behind objects placed by the timeline (even if they were
+                // both placed in the same frame to begin with).
+                dobj.base_mut(uc.gc_context).set_skip_next_enter_frame(true);
+            }
+        }
+
+        if let Some(MovieLoaderEventHandler::Avm2LoaderInfo(loader_info)) = event_handler {
+            let domain = uc
+                .library
+                .library_for_movie(movie.unwrap())
+                .unwrap()
+                .avm2_domain();
+            let mut activation = Avm2Activation::from_domain(uc.reborrow(), domain);
+            let mut loader = loader_info
+                .get_public_property("loader", &mut activation)
+                .map_err(|e| Error::Avm2Error(e.to_string()))?
+                .as_object()
+                .unwrap()
+                .as_display_object()
+                .unwrap()
+                .as_container()
+                .unwrap();
+
+            // Note that we do *not* use the 'addChild' method here:
+            // Per the flash docs, our implementation always throws
+            // an 'unsupported' error. Also, the AVM2 side of our movie
+            // clip does not yet exist. Any children added inside the movie
+            // frame constructor will see an 'added' event immediately, and
+            // an 'addedToStage' event *after* the constructor finishes
+            // when we add the movie as a child of the loader.
+            loader.insert_at_index(&mut activation.context, dobj.unwrap(), 0);
+        } else if let Some(dobj) = dobj {
+            // This is a load of an image into AVM1 - add it as a child of the target clip.
+            if dobj.as_movie_clip().is_none() {
+                let mut mc = target_clip.as_movie_clip().unwrap();
+                mc.replace_with_movie(uc, Some(movie.unwrap()), None);
```

Comment:
Same here

Reply:
What other comment is this referring to?

Reply:
I believe that it is referring to `Some(movie.unwrap())`, similar as for the above comment about `Some(dobj.unwrap())`.

Reply:
Yes, this also asserts that the movie is actualy present.